org.opentosca.bus.management.service.impl/src/org/opentosca/bus/management/service/impl/collaboration/DeploymentDistributionDecisionMaker.java

```diff
@@ -0,0 +1,420 @@
+                                                                   Constants.REQUEST_TOPIC)
+                                                       .withHeader(MBHeader.REPLYTOTOPIC_STRING.toString(),
+                                                                   Constants.RESPONSE_TOPIC)
+                                                       .withHeader(MBHeader.CORRELATIONID_STRING.toString(),
+                                                                   correlationID)
+                                                       .withHeader(MBHeader.REMOTEOPERATION_STRING.toString(),
+                                                                   RemoteOperations.invokeInstanceDataMatching)
+                                                       .build();
+
+        // publish the exchange over the camel route
+        final Thread thread = new Thread(() -> {
+
+            // By using an extra thread and waiting some time before sending the request, the
+            // consumer can be started in time to avoid loosing replies.
+            try {
+                Thread.sleep(300);
+            }
+            catch (final InterruptedException e) {
+            }
+
+            Activator.producer.send("direct:SendMQTT", request);
+        });
+        thread.start();
+
+        final String callbackEndpoint = "direct:Callback-" + correlationID;
+        DeploymentDistributionDecisionMaker.LOG.debug("Waiting for response at endpoint: {}", callbackEndpoint);
+
+        // wait for a response and consume it or timeout after 10s
+        final ConsumerTemplate consumer = Activator.camelContext.createConsumerTemplate();
+        final Exchange response = consumer.receive(callbackEndpoint, 10000);
+
+        // release resources
+        try {
+            consumer.stop();
+        }
+        catch (final Exception e) {
+            DeploymentDistributionDecisionMaker.LOG.warn("Unable to stop consumer: {}", e.getMessage());
+        }
+
+        if (response != null) {
+            DeploymentDistributionDecisionMaker.LOG.debug("Received a response in time.");
+
+            // read the deployment location from the reply
+            return response.getIn().getHeader(MBHeader.DEPLOYMENTLOCATION_STRING.toString(), String.class);
+        } else {
+            DeploymentDistributionDecisionMaker.LOG.debug("No response received within the timeout interval.");
+            return null;
+        }
+    }
+
+    /**
+     * Remove the 'State' property from the given properties Map if it is defined and return the
+     * corresponding entry Set.
+     *
+     * @param properties the properties as Map
+     * @return the properties as entry Set without 'State' property
+     */
+    private static Set<Entry<String, String>> getEntrySetWithoutState(final Map<String, String> properties) {
+        return properties.entrySet().stream().filter((entry) -> !entry.getKey().equals("State"))
```

Comment:
either the code or the javadoc lies. If you want to remove `State` from the Map, use `removeIf` or `remove`. If you want a filtered entrySet, correct the javadoc